src/session/user.rs

```diff
@@ -319,4 +380,33 @@ impl User {
     pub fn status_expression(user_expression: &gtk::Expression) -> gtk::Expression {
         gtk::PropertyExpression::new(User::static_type(), Some(user_expression), "status").upcast()
     }
+
+    pub fn formatted_status_expression(user_expression: &gtk::Expression) -> gtk::Expression {
+        let type_expression = User::type_expression(user_expression);
+        let status_expression = User::status_expression(user_expression);
+        gtk::ClosureExpression::new(
+            |args| {
+                let user_status = &args[1].get::<BoxedUserStatus>().unwrap().0;
+                let type_ = &args[2].get::<BoxedUserType>().unwrap().0;
+                match type_ {
+                    UserType::Regular => match user_status {
+                        UserStatus::Online(_data) => gettext("online"),
+                        UserStatus::Offline(data) => {
+                            let unix = data.was_online;
+                            super::user::last_seen_time_ago(unix)
+                        }
+                        UserStatus::Recently => gettext("last seen recently"),
+                        UserStatus::LastWeek => gettext("last seen last week"),
+                        UserStatus::LastMonth => gettext("last seen last month"),
+                        UserStatus::Empty => gettext("last seen a long time ago"),
+                    },
+                    UserType::Bot(_data) => gettext("bot"),
+                    UserType::Deleted => gettext("last seen a long time ago"),
```

Comment:
Is this intentional? Or should it be "deleted"?

Reply:
What do you mean? the caption is the same as in the officiall client

Reply:
That was exactly what I wanted to know. I couldn't test as I have no deleted accounts.